lib/rack/session/cookie.rb

```diff
@@ -135,14 +203,36 @@ def extract_session_id(request)
 
       def unpacked_cookie_data(request)
         request.fetch_header(RACK_SESSION_UNPACKED_COOKIE_DATA) do |k|
-          session_data = request.cookies[@key]
+          cookie_data = request.cookies[@key]
+          session_data = nil
+
+          # Try to decrypt the session data with our encryptors
+          encryptors.each do |encryptor|
+            begin
+              session_data = encryptor.decrypt(cookie_data) if cookie_data
+              break
+            rescue Rack::Encryptor::Error
+              # TODO warn? should there be some sort of external signal?
```

Comment:
Logging to `request.env[rack.errors]` is probably the best way to do this, I think.